com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/resources/ShutdownResource.java

```diff
@@ -0,0 +1,106 @@
+package com.sap.cloud.lm.sl.cf.web.resources;
+
+import java.text.MessageFormat;
+import java.time.ZonedDateTime;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.shutdown.model.Shutdown;
+
+@Component
+@Scope(value = "singleton")
+@Produces(MediaType.APPLICATION_JSON)
+public class ShutdownResource {
+
+    @Inject
+    private FlowableFacade flowableFacade;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShutdownResource.class);
+
+    static final String ACTION = "shutdown";
+
+    private static final long DEFAULT_COOLDOWN_TIME_IN_SECONDS = 2100L; // 35 mins
+    private long cooldownTimeoutInSeconds = DEFAULT_COOLDOWN_TIME_IN_SECONDS;
+
+    private volatile boolean isShutdowned = false;
+    private volatile ZonedDateTime shutdownRequestDate;
+    private volatile ZonedDateTime shutdownedAtDate;
+
+    @POST
+    public Shutdown shutdown(@HeaderParam("x-cf-applicationid") String appId,
+        @HeaderParam("x-cf-instanceid") String appInstanceId, @HeaderParam("x-cf-instanceindex") String appInstanceIndex,
+        @QueryParam("cooldownTimeoutInSeconds") String cooldownTimeoutInSecondsParam) {
+
+        if (shutdownRequestDate != null) {
+            return new Shutdown.Builder().isShutdowned(isShutdowned)
+                .shutdownRequestDate(shutdownRequestDate)
+                .shutdownedAtDate(shutdownedAtDate)
+                .appId(appId)
+                .appInstanceId(appInstanceId)
+                .appInstanceIndex(appInstanceIndex)
+                .cooldownTimeoutInSeconds(cooldownTimeoutInSeconds)
+                .build();
+        }
+
+        if (cooldownTimeoutInSecondsParam != null) {
+            cooldownTimeoutInSeconds = Long.parseLong(cooldownTimeoutInSecondsParam);
+        }
+
+        CompletableFuture.runAsync(() -> {
+            flowableFacade.setSecondsToWaitOnShutdown(cooldownTimeoutInSeconds);
+            shutdownRequestDate = ZonedDateTime.now();
+            LOGGER.info(MessageFormat.format(Messages.SHUTDOWN_REQUEST, appId, appInstanceId, appInstanceIndex, shutdownRequestDate,
+                cooldownTimeoutInSeconds));
+            flowableFacade.shutdown();
+        })
+            .thenRun(() -> {
+                isShutdowned = true;
```

Comment:
I am not like :) try to use something better than a shared boolean variable.
For instance, if someone polls the shutdown status, try to get the desired information from the processEngine.

Reply:
Also, add the logic in the method which is called for the status.